src/resolver.ts

```diff
@@ -1061,6 +1061,37 @@ export class Resolver extends DiagnosticEmitter {
     return null;
   }
 
+  resolveCompareExpressionCommonType(
+    /** The expression to resolve. */
+    expression: BinaryExpression,
+    /** Contextual flow. */
+    ctxFlow: Flow
+  ): Type | null {
+    const left = expression.left;
+    const right = expression.right;
+
+    const leftType = this.resolveExpression(left, ctxFlow, Type.auto, ReportMode.Report);
+    if (!leftType) return null;
+    const rightType = this.resolveExpression(right, ctxFlow, Type.auto, ReportMode.Report);
+    if (!rightType) return null;
```

Comment:
This should make sure diagnostics are reported for both expressions like (I think it was) before:
```suggestion
    const leftType = this.resolveExpression(left, ctxFlow, Type.auto, ReportMode.Report);
    const rightType = this.resolveExpression(right, ctxFlow, Type.auto, ReportMode.Report);
    if (!leftType || !rightType) return null;
```